state/application.go

```diff
@@ -901,6 +901,10 @@ func (a *Application) DeployedMachines() ([]*Machine, error) {
 		// whether principal or subordinate.
 		id, err := u.AssignedMachineId()
 		if err != nil {
+			if errors.IsNotAssigned(err) {
```

Comment:
strictly speaking there should be a test for this

Reply:
tests added